### PR TITLE
Undertone dev

### PR DESCRIPTION
**Pull request by omerko**
- fix adapter system tests
- fix lint issues
- change placement parameter to optional

### CODE REVIEW

File: undertone-htb.js [hunk 1]
```diff
@@ -306,84 +310,116 @@ function UndertoneHtb(configs) {
...
 
             returnParcel.targetingType = 'slot';
             returnParcel.targeting = {};
-            returnParcel.targeting[__baseClass._configs.targetingKeys.id] = [returnParcel.requestId];
             returnParcel.pass = false;
 
-            var currAdResponse = adResponseDic[returnParcel.requestId];
-            if (currAdResponse == null) {
+            var currAdResponse = adResponseDic[returnParcel.xSlotRef.bidId];
+            if (currAdResponse === null) {
```
Comment (ix-certification) [inline undertone-htb.js]: This condition will never equate to 'true' in the scenario where `adResponseDic` does not have the expected property `returnParcel.xSlotRef.bidId` as a result `currAdResponse` will not be instantiated as an empty object. In order to ensure that this works as expected please `update === null` to `=== undefined`.

However, if a valid response (bid as well as pass) from Undertone's platform always has all the `bidRequestId`, that were sent to it in a single request, in the response then this wouldn't impact the functioning of the adapter.

File: undertone-htb.js [hunk 2]
```diff
@@ -306,84 +310,116 @@ function UndertoneHtb(configs) {
...
                 currAdResponse = {};
             }
 
-            if (currAdResponse.ad == null || currAdResponse.cpm <= 0) {
+            if (currAdResponse.ad === null || currAdResponse.cpm <= 0) {
```
Comment (ix-certification) [inline undertone-htb.js]: This `currAdResponse.ad === null` should be updated to `currAdResponse.ad === undefined` as the property `ad` will be undefined in the case where there was no match for `currAdResponse`

File: undertone-htb.js [hunk 3]
```diff
@@ -306,84 +310,116 @@ function UndertoneHtb(configs) {
          *
          */
 
-        /* ---------- Process adResponse and extract the bids into the bids array ------------*/
-
         var adResponseDic = {};
-        if (adResponse != null) {
-            for (var i=0; i<adResponse.length; i++) {
-                adResponseDic[adResponse[i].bidRequestId] = adResponse[i];
+        if (adResponse !== null) {
```
Comment (ix-certification) [inline undertone-htb.js]: The `null` here needs to be updated to `undefined` to ensure we are comparing the type and value appropriately.

File: undertone-htb.js [hunk 4]
```diff
@@ -306,84 +310,116 @@ function UndertoneHtb(configs) {
...
-            returnParcel.adm = currAdResponse.ad || "";
-            returnParcel.price = currAdResponse.cpm || 0;
+            var bidCreative = currAdResponse.ad || '';
+            var bidPrice = currAdResponse.cpm || 0;
+            var bidDealId = '';
+            var pixelUrl = '';
 
-            if (currAdResponse.width != null) {
-                returnParcel.size = [currAdResponse.width, currAdResponse.height];
+            if (currAdResponse.width !== null) {
```
Comment (ix-certification) [inline undertone-htb.js]: The `null` here needs to be updated to `undefined` to ensure we are comparing the type and value appropriately.

**Review by ix-certification — COMMENTED** [review]
After doing the final round of review we found these last set of issues that need to be addressed before the code can be merged in.

File: undertone-htb.js (hunk 1 above)
Comment (omerko) [reply undertone-htb.js]: I can't do that because when there's your lint check does not allow to write undefined

File: undertone-htb.js (hunk 2 above)
Comment (omerko) [reply undertone-htb.js]: I can't do that because when there's your lint check does not allow to write undefined

File: undertone-htb.js (hunk 3 above)
Comment (omerko) [reply undertone-htb.js]: I can't do that because when there's your lint check does not allow to write undefined

File: undertone-htb.js (hunk 4 above)
Comment (omerko) [reply undertone-htb.js]: I can't do that because when there's your lint check does not allow to write undefined

File: undertone-htb.js (hunk 2 above)
Comment (ix-certification) [reply undertone-htb.js]: Please use `typeof currAdResponse.ad === 'undefined'` instead then.

File: undertone-htb.js (hunk 1 above)
Comment (ix-certification) [reply undertone-htb.js]: Please use `typeof currAdResponse === 'undefined'` instead then.

File: undertone-htb.js (hunk 4 above)
Comment (ix-certification) [reply undertone-htb.js]: Please use typeof `currAdResponse.width === 'undefined'` instead then.

Comment (omerko) [reply undertone-htb.js]: when I do that it makes system tests fail for some reason, also fails when I check both null and undefined

Comment (ix-certification) [reply undertone-htb.js]: We were able to reproduce and identify why this was failing. Please update the `ad` property in  `getValidResponse` from `ad: creative` to `ad: creative || '<div id="1"></div>'`

Comment (omerko) [reply undertone-htb.js]: ok it solved one of two system tests failures.
Still fails this test:
System Tests: Partner Open Market Bids should log headerstats events correctly
Expected 1 to be 2.

Comment (ix-certification) [reply undertone-htb.js]: Did we update both instances of `ad: creative` ?

Comment (omerko) [reply undertone-htb.js]: You're right, fixed them both and now it works, changes pushed

**Pull request merged**